core/solver/batch_bicgstab_kernels.hpp

```diff
@@ -90,6 +90,103 @@ inline int local_memory_requirement(const int num_rows, const int num_rhs)
 }
 
 
+struct storage_config {
+    // preconditioner storage
+    bool prec_shared;
+    // total number of shared vectors
+    int n_shared;
+    // number of vectors in global memory
+    int n_global;
+    // global stride from one batch entry to the next
+    int gmem_stride_bytes;
```

Comment:
is it necessary to store this in bytes? In the instances I've seen it used, it was always divided again by the value type size

Reply:
But I think as we want to align to a certain byte size, it is cleaner to store it in bytes and divide by value_type size as necessary.